debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsSnapshotProducerIT.java

```diff
@@ -480,6 +487,118 @@ public void shouldNotSnapshotMaterializedViews() throws Exception {
         consumer.process(record -> assertReadRecord(record, expectedValueByTopicName));
     }
 
+    @Test
+    @FixFor("DBZ-1413")
+    public void shouldSnapshotDomainTypeWithPropagatedSourceTypeAttributes() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.execute("CREATE DOMAIN float83 AS numeric(8,3) DEFAULT 0.0;");
+        TestHelper.execute("CREATE DOMAIN money2 AS MONEY DEFAULT 0.0;");
+        TestHelper.execute("CREATE TABLE alias_table (pk SERIAL, salary money, salary2 money2, a numeric(8,3), area float83, PRIMARY KEY(pk));");
+        TestHelper.execute("INSERT INTO alias_table (salary, salary2, a, area) values (7.25, 8.25, 12345.123, 12345.123);");
+
+        buildNoStreamProducer(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.DECIMAL_HANDLING_MODE, DecimalHandlingMode.DOUBLE)
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with("column.propagate.source.type", "public.alias_table.area,public.alias_table.a"));
+
+        final TestConsumer consumer = testConsumer(1, "public");
+        consumer.await(TestHelper.waitTimeForRecords() * 30, TimeUnit.SECONDS);
+
+        // Specifying alias money2 results in JDBC type '2001' for 'salary2'
+        // Specifying money results in JDBC type '8' for 'salary'
+
+        List<SchemaAndValueField> expected = Arrays.asList(
+                new SchemaAndValueField("salary", Decimal.builder(2).optional().build(), BigDecimal.valueOf(7.25)),
+                new SchemaAndValueField("salary2", Decimal.builder(2).optional().build(), BigDecimal.valueOf(8.25)),
```

Comment:
Wouldn't it also be good to assert the schema parameters for `salary2`?

Reply:
Btw. in the docs it should somewhere say that the base type and its constraints are exposed in the schema parameters, there's no visibility of the domain type's name IIUC.